src/jobs/publish_s3.yml

```diff
@@ -0,0 +1,35 @@
+description: Publish files to S3
+
+executor: aws
+
+parameters:
+  public:
+    type: boolean
+    description: Whether to store in the public bucket
+  path:
+    type: string
+    description: Path to store the files in S3
+  files:
+    type: string
+    description: Files to publish separated by space
+
+steps:
+  - attach_workspace:
+      at: ~/workdir
+
+  - run_aws:
+      cmd_name: Send files to S3
+      cmd: |
+        if [[ << parameters.public >> ]]; then
+          bucket=$AWS_PUBLIC_ARTIFACTS_BUCKET
+        else
+          bucket=$AWS_PRIVATE_ARTIFACTS_BUCKET
+        fi
+
+        for file in << parameters.files >> ; do
+          filename=$(basename $file)
+          AWS_PROFILE=production aws s3 cp $file $bucket/<< parameters.path >>/$(cat .version)/$filename
+        done
+        if $(grep -q -E '^[0-9]+\.[0-9]+\.[0-9]+$' .version); then
+          cat .version | tr -d '\n' | AWS_PROFILE=production aws s3 cp - $bucket/<< parameters.path >>/latest
```

Comment:
so, if the condition is true, we want to do what exactly? Why do we want / need something for 'latest'? 

Imo, if we want to use S3, we should avoid to create things like this "latest", and then assume it is a tradeoff that we need to specify the version that we want to download. And if we have some very specific scenarios where we might want the latest, can't we try to find it at fetch time? Getting the latest, by time / alphabetically?

Reply:
We have only a use case for this and it's the codacy-coverage-reporter script that needs the latest version to let people to get latest bugfixes. The alternative is to do semantic versioning sorting in bash which is error prone and more hackish than this.